crates/frontend/src/utils.rs

```diff
@@ -343,3 +349,66 @@ pub fn get_config_value(
         }
     }
 }
+
+/********* Request Utils **********/
+
+use once_cell::sync::Lazy;
+static HTTP_CLIENT: Lazy<reqwest::Client> = Lazy::new(|| reqwest::Client::new());
+
+pub async fn request<'a, T, R>(
+    url: String,
+    method: reqwest::Method,
+    body: Option<T>,
+    headers: &[(&str, &str)],
+) -> anyhow::Result<R>
+where
+    T: serde::Serialize,
+    R: serde::de::DeserializeOwned,
+{
+    let parsed_headers = headers
+        .into_iter()
+        .map(|(name, value)| {
+            let header_name = HeaderName::from_str(name);
+            let header_value = HeaderValue::from_str(value);
+            match (header_name, header_value) {
+                (Ok(name), Ok(value)) => Some((name, value)),
+                _ => None,
+            }
+        })
+        .collect::<Option<Vec<(HeaderName, HeaderValue)>>>()
+        .ok_or(anyhow!("failed to parse headers"))?;
+
+    let header_map = HeaderMap::from_iter(parsed_headers);
+
+    let mut request_builder =
+        HTTP_CLIENT.request(method.clone(), url).headers(header_map);
+    request_builder = match (method, body) {
+        (reqwest::Method::GET | reqwest::Method::DELETE, _) => request_builder,
```

Comment:
Import reqwest::Method above.

Reply:
Don't really think we need to import, this is good enough.

Reply:
import this would make this more concise and cleaner, not mandatory, just a recommendation.